llvm-spirv/lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4883,43 +4886,34 @@ bool LLVMToSPIRVBase::transExecutionMode() {
         }
       } break;
       case spv::ExecutionModeNoGlobalOffsetINTEL: {
-        if (BM->isAllowedToUseExtension(
-                ExtensionID::SPV_INTEL_kernel_attributes)) {
-          BF->addExecutionMode(BM->add(
-              new SPIRVExecutionMode(BF, static_cast<ExecutionMode>(EMode))));
-          BM->addExtension(ExtensionID::SPV_INTEL_kernel_attributes);
-          BM->addCapability(CapabilityKernelAttributesINTEL);
-        }
+        if (!BM->isAllowedToUseExtension(
```

Comment:
Why is this change required? Nesting level does not look too bad.

Reply:
The PR is just a cherry-pick of https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1177 .

There I made this change looking through the code back then (now I prefer not to mix up clean up changes within non-NFC patch) for two reasons:
1. LLVM code style recommends to do nesting like this if possible;
2. It brings minor (1 or 2 clocks) performance gains, since, I believe in general cases when SPV_INTEL_kernel_attributes is not supported more than those where it is supported.